src/header.js

```diff
@@ -5,53 +5,35 @@ class Header extends Component {
   constructor(props) {
     super(props);
     this.state = {
-      searchActive: "",
+      searchActive: false,
       searchText: "",
     };
 
-    this.handleDiggClick = this.handleDiggClick.bind(this);
-    this.handleRedditClick = this.handleRedditClick.bind(this);
-    this.handleMashableClick = this.handleMashableClick.bind(this);
-    this.handleFeedrClick = this.handleFeedrClick.bind(this);
+    this.selectFeed = this.selectFeed.bind(this);
     this.handleSearchClick = this.handleSearchClick.bind(this);
     this.handleSearchChange = this.handleSearchChange.bind(this);
   }
 
-  handleSelectSource() {
-    this.setState({ selectSource:  "active" });
-
-  }
-
-  handleDiggClick() {
-    this.props.onSelect("Digg");
-  }
-
-  handleRedditClick() {
-    this.props.onSelect("Reddit");
-  }
-
-  handleMashableClick() {
-    this.props.onSelect("Mashable");
-  }
-
-  handleFeedrClick() {
-    this.props.onSelect("All");
+  selectFeed(feed) {
+    return () => {
+      this.props.onSelect(feed);
+    }
```

Comment:
You can DRY all of the handleXClick functions with a function that looks like this.

You could also DRY it up by using anonymous functions directly in your onClick calls for example: `onClick={() => this.props.onSelect("Digg")}`